app/quest-boost/claim/[boostId]/page.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import styles from "@styles/questboost.module.css";
+import { CallData, uint256 } from "starknet";
+import { getBoostById, getQuestBoostClaimParams } from "@services/apiService";
+import { useAccount } from "@starknet-react/core";
+import Button from "@components/UI/button";
+import { useNotificationManager } from "@hooks/useNotificationManager";
+import { NotificationType, TransactionType } from "@constants/notifications";
+import { CDNImage } from "@components/cdn/image";
+import Lottie from "lottie-react";
+import verifiedLottie from "@public/visuals/sq_claim.json";
+import { hexToDecimal } from "@utils/feltService";
+
+type BoostQuestPageProps = {
+  params: {
+    boostId: string;
+  };
+};
+
+export default function Page({ params }: BoostQuestPageProps) {
+  const { address, account } = useAccount();
+  const { boostId } = params;
+  const [boost, setBoost] = useState<Boost>();
+  const [sign, setSign] = useState<Signature>(["", ""]);
+  const { addTransaction } = useNotificationManager();
+  const [displayCard, setDisplayCard] = useState<boolean>(false);
+  const [displayLottie, setDisplayLottie] = useState<boolean>(true);
+
+  const fetchPageData = async () => {
+    const boostInfo = await getBoostById(boostId);
+    setBoost(boostInfo);
+  };
+
+  useEffect(() => {
+    fetchPageData();
+  }, []);
+
+  const fetchBoostClaimParams = async (): Promise<Signature> => {
+    let formattedSign: Signature = ["", ""];
+    try {
+      if (!boost?.id || !address) return formattedSign;
+      const res = await getQuestBoostClaimParams(boost.id);
+      formattedSign = [res?.r, res?.s];
+      return formattedSign;
+    } catch (err) {
+      console.log(err);
+      return formattedSign;
+    }
+  };
+
+  const handleClaimClick = async () => {
+    const signature = await fetchBoostClaimParams();
+    if (!signature) return;
+    setSign(signature);
+  };
+
+  useEffect(() => {
+    if (!account || !boost || sign[0].length === 0 || sign[1].length === 0)
+      return;
+
+    const callContract = async () => {
+      const amount = uint256.bnToUint256(boost.amount);
+      const claimCallData = CallData.compile({
```

Comment:
Can you use a `boostCallData` file like we now do on the app ? Same architecture please :) 

Example here https://github.com/starknet-id/app.starknet.id/blob/testnet/utils/callData/registrationCalls.ts

Reply:
done